lib/spack/spack/environment.py

```diff
@@ -1057,6 +1232,48 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             activate(self._previous_active)
 
 
+def _concretize_from_constraints(spec_constraints):
```

Comment:
This is not beautiful but I know we're going to follow up immediately with improvements so I'll just leave it for now.